### PR TITLE
AK: De-`DeprecatedString`-ize `UUID` and Base64 encoding

### PR DESCRIPTION
**Pull request by GMTA**
See commits!

### CODE REVIEW

File: AK/UUID.cpp [hunk 1]
```diff
@@ -109,7 +109,7 @@ DeprecatedString UUID::to_deprecated_string() const
     builder.append(encode_hex(m_uuid_buffer.span().slice(8).trim(2)).view());
     builder.append('-');
     builder.append(encode_hex(m_uuid_buffer.span().slice(10).trim(6)).view());
```
Comment (Hendiadyoin1) [inline AK/UUID.cpp:111]: For this line and the above, i think builder has a try_append

(Also I am sure there is a better way to slice with an offset than span-slice-trim-view)

File: Userland/Libraries/LibCrypt/crypt.cpp [hunk 2]
```diff
@@ -52,8 +52,13 @@ char* crypt_r(char const* key, char const* salt, struct crypt_data* data)
...
 
     auto digest = sha.digest();
-    auto string = encode_base64(ReadonlyBytes(digest.immutable_data(), digest.data_length()));
+    auto string_or_error = encode_base64({ digest.immutable_data(), digest.data_length() });
+    if (string_or_error.is_error()) {
+        errno = ENOMEM;
+        return nullptr;
+    }
 
+    auto string = string_or_error.release_value().bytes_as_string_view();
```
Comment (ADKaster) [inline Userland/Libraries/LibCrypt/crypt.cpp:61]: this smells like use after free. If you use `.value()` instead, it won't be.

Comment (GMTA) [reply Userland/Libraries/LibCrypt/crypt.cpp]: Good point, fixed!

File: AK/UUID.cpp (hunk 1 above)
Comment (GMTA) [reply AK/UUID.cpp]: Changed to `try_append`, modified the code to be _somewhat_ better to look at. This function is not used at all in our codebase by the way :^)

File: Userland/Libraries/LibWeb/Fetch/Fetching/Fetching.cpp [hunk 3]
```diff
@@ -1305,7 +1305,8 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<PendingResponse>> http_network_or_cache_fet
                 //    true, set authorizationValue to httpRequest’s current URL, converted to an `Authorization` value.
                 else if (http_request->current_url().includes_credentials() && is_authentication_fetch == IsAuthenticationFetch::Yes) {
                     auto const& url = http_request->current_url();
-                    authorization_value = encode_base64(DeprecatedString::formatted("{}:{}", url.username(), url.password()).bytes());
+                    auto payload = DeprecatedString::formatted("{}:{}", url.username(), url.password()).bytes();
```
Comment (GMTA) [inline Userland/Libraries/LibWeb/Fetch/Fetching/Fetching.cpp:1308]: Note to self: move `.bytes()` down

**Pull request merged**